packages/framework-core/src/helpers/retrier.ts

```diff
@@ -0,0 +1,32 @@
+import { Class, Logger } from '@boostercloud/framework-types'
+
+export async function retryIfError<TReturn>(
+  logger: Logger,
+  logicToRetry: () => Promise<TReturn>,
+  errorClassThatRetries: Class<Error>,
+  maxRetries = 1000
+): Promise<TReturn> {
+  let tryNumber
+  let errorAfterMaxTries: Error | undefined
+  for (tryNumber = 1; tryNumber <= maxRetries; tryNumber++) {
+    try {
+      logger.debug(`[retrier] Try number ${tryNumber}`)
+      const result = await logicToRetry()
+      logger.debug(`[retrier] Logic succeeded after ${tryNumber} retries`)
+      return result
+    } catch (e) {
+      checkRetryError(e, errorClassThatRetries, logger)
+      errorAfterMaxTries = e
+    }
+  }
+  throw new Error(
+    `[retrier] Reached the maximum number of retries (${maxRetries}), but still getting the following error: ${errorAfterMaxTries}`
+  )
```

Comment:
In future releases, I think that it could make sense to put the failed job in a dead letter queue instead of just failing so devs can inspect and maybe fix some of the problems

Reply:
This change is important and is part of the error handling we had in mind. Nice!

Reply:
I agree. As part of the error handler task, we should have an entity (or anything that acts as a stream of events) to store events that represent the errors we get.

This way developers can not only inspect those problems, but they can even subscribe to them and get then in real time

Reply:
Anyway! I'm writing integration tests to prove this (I already did manual tests and checked it)

Reply:
Integration tests created and proven that they fail if no optimistic concurrency is used